<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>500</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>750</width>
    <height>500</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>750</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Frp客户端</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>D:/桌面/main.ico</normaloff>D:/桌面/main.ico</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_10">
    <item row="0" column="0">
     <widget class="QTabWidget" name="page">
      <property name="minimumSize">
       <size>
        <width>732</width>
        <height>482</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QTabBar::tab{height:25}
QTabBar::tab{width:150}</string>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <property name="iconSize">
       <size>
        <width>16</width>
        <height>16</height>
       </size>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideMiddle</enum>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>状态</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QWidget" name="widget" native="true">
          <layout class="QGridLayout" name="gridLayout_3">
           <property name="leftMargin">
            <number>2</number>
           </property>
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <property name="spacing">
            <number>3</number>
           </property>
           <item row="0" column="2">
            <widget class="QComboBox" name="frpc_choose_server">
             <property name="minimumSize">
              <size>
               <width>250</width>
               <height>24</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>250</width>
               <height>24</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="5">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>请选择服务器：</string>
             </property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QPushButton" name="frpc_switch">
             <property name="minimumSize">
              <size>
               <width>120</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>120</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>开启</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QTextBrowser" name="frpc_output">
          <property name="font">
           <font>
            <family>宋体</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Sunken</enum>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QTableWidget" name="frpc_portal_list_table">
          <property name="styleSheet">
           <string notr="true">QTableWidget::item{border-bottom-width: 0.8px;border-style: solid;border-color:rgb(255, 170, 255)}
QHeaderView::section{border-bottom-width: 1.5px;border-style: solid;border-color:rgb(255, 170, 255)}
QHeaderView::section{border-right-width: 1.0px;border-style: solid;border-color:rgb(255, 170, 255)}
QTableWidget{outline:none;}
QTableWidget::item{background:rgb(0, 255, 255)}
QTableWidget::item:selected{background:rgb(85, 170, 255);}
QHeaderView::section{color:rgb(0, 85, 255); font-family: &quot;宋体&quot;; font-size: 14px; height: 24px;}</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showGrid">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>服务器</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="0">
         <widget class="QGroupBox" name="server_groupBox">
          <property name="title">
           <string>服务器列表</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>9</number>
           </property>
           <item row="0" column="0">
            <widget class="QTableWidget" name="server_list_table">
             <property name="layoutDirection">
              <enum>Qt::LeftToRight</enum>
             </property>
             <property name="styleSheet">
              <string notr="true">QTableWidget::item{border-bottom-width: 0.8px;border-style: solid;border-color:rgb(255, 170, 255)}
QHeaderView::section{border-bottom-width: 1.5px;border-style: solid;border-color:rgb(255, 170, 255)}
QHeaderView::section{border-right-width: 1.0px;border-style: solid;border-color:rgb(255, 170, 255)}
QTableWidget{outline:none;}
QTableWidget::item{background:rgb(0, 255, 255)}
QTableWidget::item:selected{background:rgb(85, 170, 255);}
QHeaderView::section{color:rgb(0, 85, 255); font-family: &quot;宋体&quot;; font-size: 16px; height: 32px;}</string>
             </property>
             <property name="locale">
              <locale language="Cherokee" country="UnitedStates"/>
             </property>
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="dragDropMode">
              <enum>QAbstractItemView::NoDragDrop</enum>
             </property>
             <property name="textElideMode">
              <enum>Qt::ElideMiddle</enum>
             </property>
             <property name="showGrid">
              <bool>false</bool>
             </property>
             <property name="gridStyle">
              <enum>Qt::DashDotLine</enum>
             </property>
             <property name="sortingEnabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QWidget" name="widget_4" native="true">
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <item row="0" column="2">
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="7">
            <widget class="QPushButton" name="server_refresh_pushButton">
             <property name="minimumSize">
              <size>
               <width>35</width>
               <height>35</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>35</width>
               <height>35</height>
              </size>
             </property>
             <property name="text">
              <string>刷新</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="server_add_pushButton">
             <property name="minimumSize">
              <size>
               <width>130</width>
               <height>35</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>130</width>
               <height>35</height>
              </size>
             </property>
             <property name="text">
              <string>添加服务器</string>
             </property>
            </widget>
           </item>
           <item row="0" column="5">
            <widget class="QPushButton" name="server_edit_pushButton">
             <property name="minimumSize">
              <size>
               <width>130</width>
               <height>35</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>130</width>
               <height>35</height>
              </size>
             </property>
             <property name="text">
              <string>修改服务器信息</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="3">
            <widget class="QPushButton" name="server_remove_pushButton">
             <property name="minimumSize">
              <size>
               <width>130</width>
               <height>35</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>130</width>
               <height>35</height>
              </size>
             </property>
             <property name="text">
              <string>移除服务器</string>
             </property>
            </widget>
           </item>
           <item row="0" column="8">
            <spacer name="horizontalSpacer_8">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="6">
            <spacer name="horizontalSpacer_7">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="4">
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>隧道</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>隧道列表</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>9</number>
           </property>
           <item row="0" column="0">
            <widget class="QTableWidget" name="portal_list_table">
             <property name="styleSheet">
              <string notr="true">QTableWidget::item{border-bottom-width: 0.8px;border-style: solid;border-color:rgb(255, 170, 255)}
QHeaderView::section{border-bottom-width: 1.5px;border-style: solid;border-color:rgb(255, 170, 255)}
QHeaderView::section{border-right-width: 1.0px;border-style: solid;border-color:rgb(255, 170, 255)}
QTableWidget{outline:none;}
QTableWidget::item{background:rgb(0, 255, 255)}
QTableWidget::item:selected{background:rgb(85, 170, 255);}
QHeaderView::section{color:rgb(0, 85, 255); font-family: &quot;宋体&quot;; font-size: 15px; height: 28px;}</string>
             </property>
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="showGrid">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QWidget" name="widget_6" native="true">
          <layout class="QGridLayout" name="gridLayout_7">
           <property name="leftMargin">
            <number>2</number>
           </property>
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <property name="spacing">
            <number>3</number>
           </property>
           <item row="0" column="5">
            <widget class="QPushButton" name="portal_remove_pushButton">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>移除隧道</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <spacer name="horizontalSpacer_9">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="6">
            <widget class="QPushButton" name="portal_edit_pushButton">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>修改隧道</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_3">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>请选择服务器：</string>
             </property>
            </widget>
           </item>
           <item row="0" column="8">
            <widget class="QPushButton" name="portal_refresh_pushButton">
             <property name="minimumSize">
              <size>
               <width>35</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>35</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>刷新</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QComboBox" name="portal_choose_server">
             <property name="minimumSize">
              <size>
               <width>240</width>
               <height>24</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>240</width>
               <height>24</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="0" column="9">
            <spacer name="horizontalSpacer_11">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="0">
            <spacer name="horizontalSpacer_10">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="4">
            <widget class="QPushButton" name="portal_add_pushButton">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>80</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>添加隧道</string>
             </property>
            </widget>
           </item>
           <item row="0" column="7">
            <spacer name="horizontalSpacer_12">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>设置</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QWidget" name="widget_2" native="true">
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="1" column="0">
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="0">
            <widget class="QWidget" name="widget_3" native="true">
             <property name="maximumSize">
              <size>
               <width>700</width>
               <height>16777215</height>
              </size>
             </property>
             <layout class="QGridLayout" name="gridLayout_9">
              <item row="0" column="2">
               <widget class="QPushButton" name="set_Frpc_path_pushbutton">
                <property name="minimumSize">
                 <size>
                  <width>25</width>
                  <height>25</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>25</width>
                  <height>25</height>
                 </size>
                </property>
                <property name="text">
                 <string>-&gt;</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="settings_frpc_path"/>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Frpc.exe路径：</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="font">
              <font>
               <family>Google Sans</family>
               <pointsize>10</pointsize>
               <underline>false</underline>
              </font>
             </property>
             <property name="text">
              <string>僧游云隐寺,寺隐云游僧。  —By Ruimix   2022.08.21</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_5" native="true">
          <layout class="QGridLayout" name="gridLayout_12">
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="MyLabel" name="about">
             <property name="font">
              <font>
               <underline>true</underline>
              </font>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">color:red;</string>
             </property>
             <property name="text">
              <string>关于本软件 &gt;&gt;</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="MyLabel" name="go_to_github">
             <property name="font">
              <font>
               <underline>true</underline>
              </font>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">color:red;</string>
             </property>
             <property name="text">
              <string>前往Github &gt;&gt;</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="label_7">
             <property name="styleSheet">
              <string notr="true">color:blue;</string>
             </property>
             <property name="text">
              <string>版本：1.0(2022.09.11)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyLabel</class>
   <extends>QLabel</extends>
   <header>mylabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
